src/setfit/modeling.py

```diff
@@ -33,6 +72,92 @@ def __init__(self, model, max_seq_length: int, add_normalization_layer: bool) ->
             self.model._modules["2"] = models.Normalize()
 
 
+class SetFitHead(models.Dense):
+    """
+    A SetFit head that supports binary/multi-classes logistic regression
+    for end-to-end training.
+
+    To be compatible with Sentence Transformers, we inherit `Dense` from:
+    https://github.com/UKPLab/sentence-transformers/blob/master/sentence_transformers/models/Dense.py
+
+    Args:
+        in_features (`int`, *optional*):
+            The embedding dimension from the output of the SetFit body. If ignore, will use LazyLinear.
+        out_features (`int`, defaults to `1`):
+            The number of targets.
+        temperature (`float`):
+            A logits' scaling factor when using multi-targets (i.e., number of targets more than 1).
+        bias (`bool`, *optional*, defaults to `True`):
+            Whether to add bias to the head.
+    """
+
+    def __init__(
+        self,
+        in_features: Optional[int] = None,
+        out_features: int = 1,
+        temperature: float = 1.0,
+        bias: bool = True,
+    ) -> None:
+        super(models.Dense, self).__init__()  # init on models.Dense's parent: nn.Module
+
+        self.linear = None
+        if in_features is not None:
+            self.linear = nn.Linear(in_features, out_features, bias=bias)
+        else:
+            self.linear = nn.LazyLinear(out_features, bias=bias)
+
+        self.in_features = in_features
+        self.out_features = out_features
+        self.temperature = temperature
+        self.bias = bias
+
+    def _forward(self, x: torch.Tensor) -> torch.Tensor:
+        logits = self.linear(x)
+
+        outputs = None
+        if self.out_features == 1:  # only has one target
+            outputs = torch.sigmoid(logits)
+        else:  # multiple targets
+            outputs = nn.functional.softmax(logits / self.temperature)
+
+        return outputs
+
+    def forward(self, features: Dict[str, torch.Tensor]) -> Dict[str, torch.Tensor]:
+        features.update({"prediction": self._forward(features["sentence_embedding"])})
```

Comment:
I'm curious about the need for separate `_forward()` and `forward()` methods - is there a reason we can't wrap the logic in a single method?

Reply:
My original intend is to separate the usages. So `forward()` is for being compatible with Sentence-Transformer, and on the other hand, `_forward()` can be used when users feed embeddings only.

But ya, I think I can merge them together and act correspondly by detecting what format users feed into it. Will fix it in the next commit!

Reply:
Fixed!